src/bare_metal_billing/models.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass, field
+from typing import Annotated
+from typing_extensions import Self
+import datetime
+
+import pydantic
+
+
+def parse_date(v: str | datetime.date) -> datetime.date:
+    if isinstance(v, str):
+        return datetime.datetime.fromisoformat(v)
+    return v
+
+
+def parse_bm_accelerator(v: list) -> int:
+    return len(v)
+
+
+PositiveInt = Annotated[int, pydantic.Field(gt=0)]
+DateField = Annotated[datetime.datetime, pydantic.BeforeValidator(parse_date)]
+
+
+class BMResourceProperties(pydantic.BaseModel):
+    cpu_count: Annotated[int, pydantic.Field(alias="cpus", gt=-1)] = 0
```

Comment:
For all of the instance in which you're using `typing.Annotated` with `pydantic.Field`, you could write instead:

```
cpu_count: int = pydantic.Field(alias='cpus', default=0, gt=-1)
```

I guess I don't have a strong preference here, but I wanted to make sure you were aware that using `Annotated` was not required in these situations.

Reply:
What are all the field aliases for here?  This means that you *set* a value like this:

```
>>> x = BMResourceProperties(cpus=2)
```

But then access the value like this:

```
>>> x.cpu_count
2
```

Why not just drop the alias and pick a field name that you'll both when getting and setting the value?

Reply:
@larsks I just wanted conformity between `cpu_count` and `gpu_count`. I thought the names were more explicit

Reply:
I think the alias is confusing, but I don't feel very strongly about it. Go with your preference.